Zend/zend_dimension_handlers.h

```diff
@@ -0,0 +1,46 @@
+/*
+   +----------------------------------------------------------------------+
+   | Zend Engine                                                          |
+   +----------------------------------------------------------------------+
+   | Copyright (c) Zend Technologies Ltd. (http://www.zend.com)           |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Author: Gina Peter Banyard <girgias@php.net>                         |
+   +----------------------------------------------------------------------+
+*/
+
+/* This really should not be needed... */
+#ifndef ZEND_DIMENSION_HANDLERS_H
+#define ZEND_DIMENSION_HANDLERS_H
+
+//BEGIN_EXTERN_C()
+
+#include "zend_types.h"
+
+typedef struct _zend_class_dimensions_functions {
+	/* rv is a slot provided by the callee that is returned */
+	zval *(*/* const */ read_dimension)(zend_object *object, zval *offset, zval *rv);
+	bool  (*/* const */ has_dimension)(zend_object *object, zval *offset);
+	zval *(*/* const */ fetch_dimension)(zend_object *object, zval *offset, zval *rv);
```

Comment:
I don't understand why `rv` is passed, and something is also returned. Just pass in `rv`, and make it return void, or if you need a success/failure determination, then return `zend_result` instead? Is there something I am missing? Some case where the `rv` isn't the exact same thing as what is returned, other than perhaps on error?

Same comment goes for `fetch_append` and `read_dimension`, I suppose.